addons/web/controllers/main.py

```diff
@@ -665,8 +665,8 @@ def create(self, master_pwd, name, lang, password, **post):
     @http.route('/web/database/duplicate', type='http', auth="none", methods=['POST'], csrf=False)
     def duplicate(self, master_pwd, name, new_name):
         try:
-            if not re.match('^[a-zA-Z][a-zA-Z0-9_]+$', new_name):
-                raise Exception(_('Invalid database name. Only alphanumerical characters and underscore are allowed.'))
+            if not re.match('^[a-zA-Z0-9][a-zA-Z0-9_.-]+$', new_name):
+                raise Exception(_('Invalid database name. Only alphanumerical characters, underscore, hyphen and dot are allowed.'))
```

Comment:
You have to "translate" them multiple times on the Python side if you want to do a working lazy translation: once when you define the constant, and again every time you use it.
Not sure I would bother. And the template needs a variation of the message anyway.
```python
# define static constant
DBNAME_WARNING = _("Invalid ...")  # translation = no-op, but term extraction OK
# then later use it in a method
raise Exception(_(DBNAME_WARNING))  # translation OK, ignored for extraction
```
Makes more sense for the regex indeed, if it's really trivial to pass.